File: 26 - Remove Duplicates from Sorted Array.cpp
/*
 26. Remove Duplicates from Sorted Array (E)

 Notes:
    T: O(n)
    S: O(1)
*/

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        int start = 0;
        for (int i = 1; i < nums.size(); i++)
        {
            if (nums[start] != nums[i])
                nums[++start] = nums[i];
        }
        return start + 1;
    }
};

File: JAVA/26 - Remove Duplicates from Sorted Array.java
/*
 26. Remove Duplicates from Sorted Array (E)
 
 Approach 1 :
 Time: O(n)
 Space: O(1)

*/

/* Approach 1 */
class Solution {
    public int removeDuplicates(int[] nums) {
        int index = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[index] != nums[i]) {
                nums[++index] = nums[i];
            }
        }
        return index + 1;
    }
}